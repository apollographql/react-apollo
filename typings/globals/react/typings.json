{
  "resolution": "main",
  "tree": {
    "src": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/ecaf8c1c3f858e4b73f508c2a7dc831561d2097f/react/react.d.ts",
    "raw": "registry:dt/react#0.14.0+20160319053454",
    "typings": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/ecaf8c1c3f858e4b73f508c2a7dc831561d2097f/react/react.d.ts"
  }
}
