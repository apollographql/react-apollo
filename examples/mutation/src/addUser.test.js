import React from 'react';
import { render, fireEvent, wait } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';

import AddUser, { ADD_USER } from './AddUser';

const request = {
  query: ADD_USER,
  variables: { username: 'peter' }
};

const mocks = [
  {
    request,
    result: {
      data: {
        createUser: {
          id: '1',
          username: 'peter',
          __typename: 'User'
        }
      }
    }
  }
];

const mocksWithError = [
  {
    request,
    error: new Error('Something went wrong')
  }
];

const waitUntilLoadingIsFinished = queryByText =>
  wait(() => {
    expect(queryByText('LOADING')).toBe(null);
  });

it('renders content if the mutation has not been called', () => {
  const { container } = render(
    <MockedProvider mocks={mocks}>
      <AddUser />
    </MockedProvider>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('fires the mutation', async () => {
  const {
    container,
    getByPlaceholderText,
    getByTestId,
    getByText,
    queryByText,
    debug
  } = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <AddUser />
    </MockedProvider>
  );

  const inputNode = getByPlaceholderText('Username');
  inputNode.defaultValue = 'peter';
  fireEvent.change(inputNode);

  const buttonNode = getByTestId('add-user-button');
  fireEvent.click(buttonNode);

  getByText('LOADING');

  await waitUntilLoadingIsFinished(queryByText);

  getByText('Created peter with id 1');
});

it('errors', async () => {
  const { getByTestId, getByText, getByPlaceholderText, queryByText } = render(
    <MockedProvider mocks={mocksWithError} addTypename={false}>
      <AddUser />
    </MockedProvider>
  );

  const inputNode = getByPlaceholderText('Username');
  inputNode.value = 'peter';
  fireEvent.change(inputNode);

  const buttonNode = getByTestId('add-user-button');
  fireEvent.click(buttonNode);

  await waitUntilLoadingIsFinished(queryByText);

  getByText('ERROR');
});
