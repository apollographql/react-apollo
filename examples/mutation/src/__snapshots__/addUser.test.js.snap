// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content if the mutation has not been called 1`] = `
<div>
  <input
    placeholder="Username"
    value=""
  />
  <button
    data-testid="add-user-button"
  >
    Create new user
  </button>
</div>
`;
